salt/modules/file.py

```diff
@@ -3078,10 +3078,14 @@ def touch(name, atime=None, mtime=None):
     '''
     name = os.path.expanduser(name)
 
-    if atime and atime.isdigit():
+    try:
         atime = int(atime)
-    if mtime and mtime.isdigit():
```

Comment:
Rather than these try/excepts (which now have bare exceptions, which we try to avoid in Salt), why don't you use something like an `isinstance` check instead of `isdigit`?

Reply:
@gurubert Did you see the comments above?